File: back/src/main/java/com/backend/back/models/Inventario.java
package com.backend.back.models;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "inventario")
@Data
public class Inventario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombre;
    private String descripcion;
    private Double precio;
    private Long stock;
    @JoinColumn(name = "categoria_id", referencedColumnName = "id")
    @ManyToOne
    private Categoria categoriaId;
    @JoinColumn(name = "proveedor_id", referencedColumnName = "id")
    @ManyToOne
    private Persona proveedorId;


}

File: frontend/src/app/modelos/inventario.ts
import {Categoria} from './Categorias';
import {Personas} from './personas';

export interface Inventario {
  id : number;
  nombre: string;
  descripcion: string;
  precio: number;
  stock: number;
  imagenRef : string;
  categoriaId: Categoria;
  proveedorId: Personas;
}
